android/anc/src/main/java/org/smartregister/fhircore/anc/data/patient/PatientRepository.kt

```diff
@@ -300,18 +301,22 @@ constructor(
   }
 
   suspend fun fetchVitalSigns(patientId: String, searchFilterString: String): Observation {
-    val searchFilter: SearchFilter =
-      when (searchFilterString) {
-        "body-weight" -> vitalSignsConfig.weightFilter!!
-        "body-height" -> vitalSignsConfig.heightFilter!!
-        "bp-s" -> vitalSignsConfig.BPSFilter!!
-        "bp-d" -> vitalSignsConfig.BPDSFilter!!
-        "pulse-rate" -> vitalSignsConfig.pulseRateFilter!!
-        "bg" -> vitalSignsConfig.bloodGlucoseFilter!!
-        "spO2" -> vitalSignsConfig.bloodOxygenLevelFilter!!
-        else ->
-          throw UnsupportedOperationException("Given filter $searchFilterString not supported")
-      }
+    var searchFilter: SearchFilter
+    vitalSignsConfig.run {
```

Comment:
Use `with` scope function instead.

Reply:
with scope function for AncOverviewConfiguration is not resolving

Reply:
```kotlin
with(vitalSignsConfig) {
      searchFilter =
        when (searchFilterString) {
          "body-weight" -> weightFilter!!
          "body-height" -> heightFilter!!
          "bmi" -> bmiFilter!!
          "bp-s" -> BPSFilter!!
          "bp-d" -> BPDSFilter!!
          "pulse-rate" -> pulseRateFilter!!
          "bg" -> bloodGlucoseFilter!!
          "spO2" -> bloodOxygenLevelFilter!!
          else ->
            throw UnsupportedOperationException("Given filter $searchFilterString not supported")
        }
    }
```

Reply:
updated